ci/ditt/main.py

```diff
@@ -125,80 +130,96 @@ def generateHTMLStatus(_htmlData, _cacheChanged):
 
-                #htmlRowTuple[HTML_TUPLE_RENDER_INDEX] = scene = line.strip()
                 scene = line.strip()
                 htmlRowTuple[HTML_TUPLE_RENDER_INDEX] = renderName
                 executor = str(NBL_PATHTRACER_EXE.absolute()) + ' -SCENE=' + scene + ' -TERMINATE'
                 subprocess.run(executor, capture_output=True)
 
-                # fail CI if the reference cache is different that current generated cache
                 if not filecmp.cmp(destinationReferenceCache, generatedReferenceCache):
+                    # fail CI if the reference cache is different that current generated cache
                     cacheChanged = True
                     CI_PASS_STATUS = False
 
+
                 anIndex = HTML_TUPLE_REFERENCE_INDEX
                 outputDiffTerminators = ['', '_albedo', '_normal', '_denoised']
                 for diffTerminator in outputDiffTerminators:
-                    diffImageFilepath = str(NBL_CI_REFERENCES_DIR.absolute()) + '/' + renderName + '/' + renderName + diffTerminator + '_diff.exr'
-                    diffValueFilepath = str(NBL_CI_REFERENCES_DIR.absolute()) + '/' + renderName + '/' + renderName + diffTerminator + '_diff.txt'
-                    executor = str(NBL_IMAGEMAGICK_EXE.absolute()) + ' compare -metric AE ' + destinationReferenceUndenoisedTargetName + diffTerminator + '.exr ' + generatedUndenoisedTargetName + diffTerminator + '.exr -compose src ' + diffImageFilepath
-                    magicProcess = subprocess.run(executor, capture_output=True)
-                    magicDecodeValue = magicProcess.stderr.decode()
+                    imageDiffFilePath = str(NBL_CI_REFERENCES_DIR.absolute()) + '/' + renderName + '/' + renderName + diffTerminator +"_diff.exr"
+                    imageRefFilepath = destinationReferenceUndenoisedTargetName + diffTerminator + '.exr'
+                    imageGenFilepath = generatedUndenoisedTargetName + diffTerminator + '.exr'
+                    
+                    #create difference image for debugging
+                    diffImageCommandParams = f' "{imageRefFilepath}" "{imageGenFilepath}" -fx "abs(u-v)" "{imageDiffFilePath}"'
+                    executor = str(NBL_IMAGEMAGICK_EXE.absolute()) + diffImageCommandParams
+                    magickDiffImgProcess = subprocess.run(executor, capture_output=False)
+
+                    #calculate the amount of pixels whose relative errors are above NBL_ERROR_THRESHOLD
+                    #logic operators in image magick return 1.0 if true, 0.0 if false 
+                    #image magick convert -compose divide does not work with HDRI, this requiring use of -fx 
+                    diffValueCommandParams = f" {imageRefFilepath} {imageGenFilepath}  -define histogram:unique-colors=true -fx \"(abs(u-v)/((0!=min(u,v))?abs(min(u,v)):1.0))>={NBL_ERROR_THRESHOLD}\" -format %c histogram:info:" 
```

Comment:
`min(u,v)>CLOSE_TO_ZERO ? ((abs(u-v)/min(u,v))>NBL_ERROR_THRESHOLD):(max(u,v)<=CLOSE_TO_ZERO)`

Then when one of the pixels was black enough, the other also needs to be black enough. 

Use `>=` for threshold compare, if I ever set it to 0.0, stuff will mark self as different even it isn't.

P.S. is this per channel or does it operate on channels?